core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -814,6 +879,49 @@ public void close() {
     }
   }
 
+  /**
+   * Selects the MasterWorkerInfo according to addresses from report capacity command input.
+   *
+   * @param addresses the address set that user passed in
+   * @param workerInfoSet the MasterWorkerInfo set to select info from
+   * @param workerNames the supported worker names
+   * @return the selected MasterWorkerInfo
+   */
+  private Set<MasterWorkerInfo>  selectInfoByAddress(Set<String> addresses,
+      Set<MasterWorkerInfo> workerInfoSet, Set<String> workerNames) {
+    Set<MasterWorkerInfo> selectedWorkerInfo = new HashSet<>();
```

Comment:
can we use Java 8's stream function to simplify the code?
https://www.mkyong.com/java8/java-8-streams-filter-examples/

Reply:
Thanks for the idea of taking use Stream. Changed to 
```java
  /**
   * Selects the MasterWorkerInfo from workerInfoSet whose host or related IP address
   * exists in addresses.
   *
   * @param addresses the address set that user passed in
   * @param workerInfoSet the MasterWorkerInfo set to select info from
   * @param workerNames the supported worker names
   */
  private Set<MasterWorkerInfo> selectInfoByAddress(Set<String> addresses,
      Set<MasterWorkerInfo> workerInfoSet, Set<String> workerNames) {
    return workerInfoSet.stream().filter(info -> {
      String host = info.getWorkerAddress().getHost();
      workerNames.add(host);

      String ip = null;
      try {
        ip = NetworkAddressUtils.resolveIpAddress(host);
        workerNames.add(ip);
      } catch (UnknownHostException e) {
        // The host may already be an IP address
      }

      if (addresses.contains(host)) {
        addresses.remove(host);
        return true;
      }

      if (ip != null) {
        if (addresses.contains(ip)) {
          addresses.remove(ip);
          return true;
        }
      }
      return false;
    }).collect(Collectors.toSet());
  }
```